src/mindist.rs

```diff
@@ -0,0 +1,274 @@
+use crate::Vec2;
+use core::cmp::Ordering::Less;
+
+// This implements the algorithm in "Computing the
+// minimum distance between two Bézier curves", Chen et al.,
+// *Journal of Computational and Applied Mathematics* 229(2009), 294-301
+
+pub fn min_dist_param(
+    bez1: &[Vec2],
+    bez2: &[Vec2],
+    u: (f64, f64),
+    v: (f64, f64),
+    epsilon: f64,
+    best_alpha: Option<f64>,
+) -> (f64, f64, f64) {
+    let n = bez1.len() - 1;
+    let m = bez2.len() - 1;
+    let (umin, umax) = u;
+    let (vmin, vmax) = v;
+    let umid = (umin + umax) / 2.0;
+    let vmid = (vmin + vmax) / 2.0;
+    let svalues: [(f64, f64, f64); 4] = [
+        (S(umin, vmin, bez1, bez2), umin, vmin),
+        (S(umin, vmax, bez1, bez2), umin, vmax),
+        (S(umax, vmin, bez1, bez2), umax, vmin),
+        (S(umax, vmax, bez1, bez2), umax, vmax),
+    ];
+    let alpha: f64 = svalues.iter().map(|(a, _, _)| *a).fold(f64::NAN, f64::min);
```

Comment:
I think `reduce` is slightly nicer than `fold` here, because you avoid the nan (there is an unwrap but it should be compiled out because the size is known).